dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -265,27 +265,46 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
     std::wstring iconFilePath{};
 
     wil::unique_prop_variant propVariantIcon;
-    // Throw in case of failure, since Icon is mandatory and we don't have a fallback!
-    THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
-    THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt == VT_EMPTY, "You must specify an app icon before calling Register().");
+    // In case there is no icon asset, do not throw. Set a default icon instead.
+    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
 
-    iconFilePath = propVariantIcon.pwszVal;
-
-    // Icon filepaths from Shell APIs have this format: <filepath>,-<index>,
-    // since .ico files can have multiple icons in the same file.
-    // NotificationController doesn't seem to support such format, so let it take the first icon by default.
-    auto iteratorForCommaDelimiter = iconFilePath.find_first_of(L",");
-    if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
+    if (propVariantIcon.vt == VT_LPWSTR && propVariantIcon.pwszVal != nullptr)
     {
-        iconFilePath.erase(iteratorForCommaDelimiter);
-    }
+        iconFilePath = propVariantIcon.pwszVal;
+
+        // Most icon filepaths from Shell APIs have this format: <filepath>,-<index>,
+        // since .ico files can have multiple icons in the same file.
+        // NotificationController doesn't seem to support such format, so let it take the first icon by default.
+        auto iteratorForCommaDelimiter = iconFilePath.find_first_of(L",");
+        if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
+        {
+            iconFilePath.erase(iteratorForCommaDelimiter);
+        }
+
+        auto iteratorForLastBackSlash = iconFilePath.find_last_of(L"\\");
+        auto iteratorForForwardSlash = iconFilePath.find_last_of(L"/");
+        THROW_HR_IF_MSG(ERROR_BAD_PATHNAME, iteratorForLastBackSlash != std::wstring::npos && iteratorForForwardSlash != std::wstring::npos,
+            "You must provide a valid icon filepath (do not combine forward and back slashes).");
+        auto iteratorForLastSlash = min(iteratorForLastBackSlash, iteratorForForwardSlash);
+
+        std::wstring fileName{ iconFilePath };
 
-    auto iteratorForFileExtension = iconFilePath.find_first_of(L".");
-    THROW_HR_IF_MSG(E_UNEXPECTED, iteratorForFileExtension == std::wstring::npos, "You must provide a valid filepath as the app icon.");
+        if (iteratorForLastSlash != std::wstring::npos)
```

Comment:
Not sure I understand this. We should chat offline on how to simplify everything here. Its way too complex IMHO

Reply:
After discussing offline, this isn't the right time to address this. I'll proceed to revert.